src/CssInliner.php

```diff
@@ -583,6 +618,9 @@ private function collateCssRules(CssDocument $parsedCss): array
             if (!$cssRule->hasAtLeastOneDeclaration()) {
                 continue;
             }
+            if (\count(\array_intersect($cssRule->getSelectors(), \array_keys($this->excludedCssSelectors)))) {
+                continue;
+            }
```

Comment:
I'm assuming the excluded selector list matches a full selector from a possibly comma-separated list, so `.example` will not exclude `p.example` or `p .example`.  It would be good to confirm though that `p .example` would also exclude selectors written with varying whitespace (e.g. newline or multiple spaces), perhaps using a `dataProvider` for the currently-named `addExcludedCssSelector` test.

Reply:
This has been taken care of here: https://github.com/MyIntervals/emogrifier/pull/1202/files#diff-d57870a239314b85bdf5a191a3629018f9c95e6019b3405584d36b3b9abddd48R638-R642

Reply:
And tested here: https://github.com/MyIntervals/emogrifier/pull/1202/files#diff-2eda2d27b064a67021c7004fd070acb00ac6bf08695521484af6c087c6eec0baR3188-R3197

Reply:
My point was just to add a couple of extra tests to help define and isolate the precise behaviour of the new functionality.  Thanks for continued engagement and persuing this.  Your idea is good.

I'll look into your latest changes in the next few days, and expect @oliverklee will too.

Thanks again <3

Reply:
> My point was just to add a couple of extra tests to help define and isolate the precise behaviour of the new functionality

And tt was totally legit, I wasn't sure it was meant to be handled on the user input (`addExcludedCssSelector('.example p)`), or on the selectors found in the provided CSS. I went for the provided CSS only since weird spaces/lines breaks or very unlikely passed in `addExcludedCssSelector` method. Let me know if you feel like it normalization should be applied on both.

Reply:
> I wasn't sure it was meant to be handled on the user input (`addExcludedCssSelector('.example p')`), or on the selectors found in the provided CSS.

Not quite sure what you mean by this.  Aren't they both user input?  In context, the most usery of input will be the CSS.  The parameter to the function, whilst also user input, while likely be more carefully chosen ;)

Reply:
> The parameter to the function, whilst also user input, will likely be more carefully chosen ;)

I think w're both on the same side here :)

Reply:
> Let me know if you feel like it normalization should be applied on both.

Sorry, I've only just fully read and understood your comment.  Though that doesn't change anything, other than that my inrigue is now piqued as to whether a programmer error in the whitespacing department is covered too (it may well be).